src/global/pidfile.cc

```diff
@@ -170,9 +170,15 @@ int pidfh::open(const ConfigProxy& conf)
     .l_len = 0
   };
   int r = ::fcntl(pf_fd, F_SETLK, &l);
-  if (r < 0) {
-    derr << __func__ << ": failed to lock pidfile "
-	 << pf_path << " because another process locked it." << dendl;
+  if (r != 0) {
+    if ( r == EAGAIN ) {
```

Comment:
i don't think `fcntl(2)` returns `EAGAIN`. probably you meant to check `errno`?

Reply:
@tchaikov 
I took the FreeBSD manpage text, but the Linux text in that sense is much more obvious:
```
F_SETLK (struct flock *)
              Acquire a lock (when l_type is F_RDLCK or F_WRLCK) or release
              a lock (when l_type is F_UNLCK) on the bytes specified by the
              l_whence, l_start, and l_len fields of lock.  If a conflicting
              lock is held by another process, this call returns -1 and sets
              errno to EACCES or EAGAIN.  (The error returned in this case
              differs across implementations, so POSIX requires a portable
              application to check for both errors.)
```
FreeBSD writes:
```
               If a shared	or exclusive lock cannot be set,
		fcntl()	returns	immediately with EAGAIN.
```
I guess they sort of mean the same...

Reply:
i don't follow you. are you suggesting that you were right in previous change? if yes, why are you updated your code per my suggestion. or what you want to approve?

i am quoting from your citation 

> this call returns -1 and sets errno to EACCES or EAGAIN

so, it does return `-1`.

and per http://man7.org/linux/man-pages/man2/fcntl.2.html

> On error, -1 is returned, and errno is set appropriately

per http://pubs.opengroup.org/onlinepubs/009695399/functions/fcntl.html

> F_SETFL
> Value other than -1.
> Otherwise, -1 shall be returned and errno set to indicate the error.

so we should check for `-1` per the POSIX standard.

Reply:
@tchaikov 
Had to look in the source to be sure. OTOH function return are "always" TRUE<>FALSE with the reason in errorcode. So why would this be different? It is just the wording that is messy.